SimG4Core/SensitiveDetector/interface/SensitiveDetectorMakerBase.h

```diff
@@ -26,20 +27,27 @@ namespace edm {
 
 class SensitiveDetectorMakerBase {
 public:
-  explicit SensitiveDetectorMakerBase(){};
-  virtual ~SensitiveDetectorMakerBase(){};
-
-  // ---------- const member functions ---------------------
-  virtual SensitiveDetector* make(const std::string& iname,
-                                  const edm::EventSetup& es,
-                                  const SensitiveDetectorCatalog& clg,
-                                  const edm::ParameterSet& p,
-                                  const SimTrackManager* man,
-                                  SimActivityRegistry& reg) const = 0;
-
-private:
+  explicit SensitiveDetectorMakerBase() = default;
+  virtual ~SensitiveDetectorMakerBase();
   SensitiveDetectorMakerBase(const SensitiveDetectorMakerBase&) = delete;
   const SensitiveDetectorMakerBase& operator=(const SensitiveDetectorMakerBase&) = delete;
+
+  virtual void beginRun(edm::EventSetup const&);
+
+  // ---------- const member functions ---------------------
+  //deprecated API
+  virtual std::unique_ptr<SensitiveDetector> make(const std::string& iname,
```

Comment:
@Dr15Jones , SensitiveDetector cannot be unique_ptr - this will provide crash at the exit. It should be siple pointer whihc will be deleted by Geant4. Because of this problem we do not destruct of Geant4 in CMSSW.

Reply:
Turns out that change doesn't actually matter since the calling site of make was putting it directly into a unique_ptr anyway

https://github.com/cms-sw/cmssw/blob/6c6436e686e1701a302202cecf4fda09adfea8bb/SimG4Core/SensitiveDetector/src/AttachSD.cc#L29

It then does a release before giving it to G4
https://github.com/cms-sw/cmssw/blob/6c6436e686e1701a302202cecf4fda09adfea8bb/SimG4Core/SensitiveDetector/src/AttachSD.cc#L35

Reply:
Sorry, I like added this some time ago. You are right.